core/src/main/scala/filodb.core/Utils.scala

```diff
@@ -38,4 +43,161 @@ object Utils extends StrictLogging {
     availableMem
   }
 
+  /**
+   * Maps sets of {@link ColumnFilter} "keys" to element "values".
+   *
+   * Convenient for use-cases that require e.g. configuration of rules against
+   *   label-value filters (filter->rule pairs). The {@link #get} method
+   *   accepts a set of label-value pairs, identifies any matching filter sets,
+   *   then returns the corresponding element.
+   *
+   * For example, consider the (filters,element) pairs:
+   *     - ({l1="v1",l2=~"v2.*"}, elt1)
+   *     - ({l1="v1",l2="v3"},    elt2)
+   *     - ({l2=~"v4.*"},         elt3)
+   *
+   * The following {label:value} sets would map to these elements:
+   *     - {l1=v1,l2=v2,l3=v3} -> elt1
+   *     - {l1=v1}             -> (None)
+   *     - {l2=v4A}            -> elt3
+   *
+   * A label-value map "matches" a set of filters when:
+   *     - all filter labels are present in the map
+   *     - every filter is matched by label in the map.
+   *
+   * NOTE: at most one {@link ColumnFilter} in a set can be non-Equals.
+   *
+   * NOTE: This implementation is intended to favor simplicity over efficiency.
+   *   Other implementations should be considered if the {@link #get} runtime
+   *   (described in the method's javadoc) is insufficient.
+   *
+   * @tparam T the type of elements to store/return.
+   * @param filterElementPairs (filter-set,element) pairs; at most one of filter-set
+   *                           can be a non-Equals filter.
+   */
+  class ColumnFilterMap[T](filterElementPairs: Iterable[(Iterable[ColumnFilter], T)]){
```

Comment:
This implementation is more complicated than originally expected, and it isn't particularly efficient.
An alternative is implemented [here](https://github.com/filodb/FiloDB/blob/517af29d61555f054eff3a170cdae538a1ade032/core/src/main/scala/filodb.core/TargetSchemaProvider.scala#L99). Its `get` runtime complexity grows logarithmically with the total count of mappings, and it might actually be easier to reason about.

There is a drawback: it doesn't currently support regex filters, but it can be updated to support prefix-only `EqualsRegex` filters (e.g. `foo=~"bar-.*"`).